packages/graphql-mini-transforms/src/document.ts

```diff
@@ -66,7 +68,24 @@ export function extractImports(rawSource: string) {
   return {imports: [...imports], source};
 }
 
-function removeUnusedDefinitions(document: DocumentNode) {
+export function toSimpleDocument<Data, Variables, DeepPartial>(
+  document: DocumentNode<Data, Variables, DeepPartial>,
+): SimpleDocument<Data, Variables, DeepPartial> {
+  return {
+    id: document.id,
```

Comment:
Would `source` be an adequate cache key?

Reply:
It is adequate but inefficient, especially for some queries that need to query enormous parts of a graph (like checkout's proposal query, or many of the online store team's queries for getting theme details).